### ISSUE TITLE
vis-network.js does not work in strict mode enforced by Chrome extensions

### ISSUE THREAD

**Issue by naskooskov** [visjs/vis-network#271]
Hello,
I've been using older version of vis.js in a Chrome extension I've written. I wanted to update to the newest version, however it looks like it does not support running in that environment:

Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' blob: filesystem:".

    at Function (<anonymous>)
    at vis-network.js:14949
    at createCommonjsModule$2 (vis-network.js:12635)
    at vis-network.js:14247
    at vis-network.js:29
    at vis-network.js:30
(anonymous) @ vis-network.js:14949
createCommonjsModule$2 @ vis-network.js:12635
(anonymous) @ vis-network.js:14247
(anonymous) @ vis-network.js:29
(anonymous) @ vis-network.js:30

I'm using Google Chrome, 80.0.3970.5 on Linux and vis.js 6.4.4. It would be great if the library can be updated to avoid unsafe-eval, which will allow it to run inside Chrome extension pages.  
Just creating a barebones extension that just includes the vis-network.jus will reproduce the issue right off the bat. I cannot easily provide that in jsbin format, but let me know if you need help with creating a simple extension and I'd be happy to help.

**Comment by Thomaash** [comment]
Hi @naskooskov,

this code actually comes from Facebook's Regenerator Runtime through Babel. I tried deleting the offending line and it seems to work just fine without it. Since it's never executed I'm not exactly sure what it's good for. It seems to be a fallback for misconfigured builds or something like that. Maybe an issue on https://github.com/facebook/regenerator/issues/new could be opened to see if they could fix it upstream? The problem is that Chrome does static analysis so even though it's dead code it prevents the whole extension from running.

The other solution is to simply add `
  "content_security_policy": "script-src 'self' 'unsafe-eval'; object-src 'self'",` into your manifest.json and enable eval.

**Comment by naskooskov** [comment]
It looks like regenerator has been fixed. Is this something that needs to be pulled into vis.js to get the updated version?

While I could relax the CSP, it is there for a good reason. As someone who works on security, I'd rather not do that.

**Comment by Thomaash** [comment]
Where do you see the fix? I would also prefer not to violate CSP but looking at their master branch the offending line is still there: https://github.com/facebook/regenerator/blob/master/packages/regenerator-runtime/runtime.js#L725.

**Comment by patleb** [comment]
@Thomaash, the problem is that there is dynamic js in this package.

**Comment by Thomaash** [comment]
Hi @naskooskov,

recently we made some changes to mitigate similar issue with Vis Timeline. Did it help here too or is it still broken in Chrome?

**Comment by wthoutanymmries** [comment]
@Thomaash I encountered the same issue with webpack CSP in my electron app. Do you have a solution?
![image](https://user-images.githubusercontent.com/19974139/89176877-c9306900-d5a3-11ea-8263-930a86efb120.png)
![image](https://user-images.githubusercontent.com/19974139/89176896-d2213a80-d5a3-11ea-9ae6-56996d965c68.png)

**Comment by Thomaash** [comment]
Hi @wthoutanymmries,

could you provide an MWE, please? Because this works just fine for me.

Thanks.

PS: Working extension [ext.zip](https://github.com/visjs/vis-network/files/5018055/ext.zip).

**Comment by wthoutanymmries** [comment]
@Thomaash Yup, everything seems to work fine but the console is full of errors (I'm intending to heavily use this library, but I just recently started developing, so I am not aware of any consequences this may produce in the future)
Here's an attachment. Just run `npm install && npm run dev`
The CSP is located in webpack.development.js

Thank you for your time!
![image](https://user-images.githubusercontent.com/19974139/89269801-d7d65900-d653-11ea-8836-e2e9f669d2bc.png)
![image](https://user-images.githubusercontent.com/19974139/89269875-f3d9fa80-d653-11ea-8483-61035c16ab61.png)
![image](https://user-images.githubusercontent.com/19974139/89270057-3d2a4a00-d654-11ea-8e3c-7e5ab82eae37.png)
[graphx.zip](https://github.com/visjs/vis-network/files/5020930/graphx.zip)

**Comment by Thomaash** [comment]
I can't try out your example right now, but the default policy as per the official docs is `script-src 'self'; object-src 'self'
`, since I didn't configure CSP in any way there is no `unsafe-eval` enabled in my example and it works.

**Comment by wthoutanymmries** [comment]
@Thomaash Ok, I've tried `script-src 'self'; object-src 'self'`, but there's the same error and the app doesn't load.
**BTW** I'am using [https://github.com/reZach/secure-electron-template](url) if it will be of any help
Thanks!
![image](https://user-images.githubusercontent.com/19974139/89329844-d4b98800-d6a8-11ea-8cb8-e6310ba0f1f6.png)

**Comment by Thomaash** [comment]
It would be best if you just sent me a built, ready to use extension with as little code as possible (i.e. without Webpack, Electron… e.g. modify the extension I sent you to manifest the issue).

**Comment by wthoutanymmries** [comment]
Ok, so I'm basically nobody. I've tried adding meta tag with policies and it did NOTHING, except for `style-src 'self'`, the console was spitting errors at me because of all the inlines. Maybe I misconfigured something (as that 
> ![image](https://user-images.githubusercontent.com/19974139/89176877-c9306900-d5a3-11ea-8263-930a86efb120.png)
comment said the module was not supposed to run in strict mode and etc etc). I'll ditch CSP altogether for now until I have better understanding of how everything works and will continue developing

Again, thank you for your time!
